### PR TITLE
plugins(key-auth): various improvements

### PR DESCRIPTION
**Pull request by jschmid1**
### Description

requesting copy-edit for

* logical circuits
* anonymous consumer
* decision table for common scenarios

Signed-off-by: Joshua Schmid <jaiks@posteo.de>

https://konghq.atlassian.net/browse/KAG-397

### Testing instructions

https://deploy-preview-5140--kongdocs.netlify.app/

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/_hub/kong-inc/key-auth/_index.md [hunk 1]
```diff
@@ -384,6 +393,19 @@ Response:
 [consumer-object]: /gateway/latest/admin-api/#consumer-object
 [acl-associating]: /plugins/acl/#associating-consumers
 
+
+### Decision table (for the lack of a better name) -> This should be included in every plugin
```
Comment (jschmid1) [inline app/_hub/kong-inc/key-auth/_index.md:397]: Suggestion for names:

* Scenario Matrix
* Behavior Matrix
* Decision Table

**Comment by cloudjumpercat** [comment]
@jschmid1 Hi! Thanks for making these updates! Are you still working on this?

File: app/_hub/kong-inc/key-auth/_index.md [hunk 2]
```diff
@@ -384,6 +392,19 @@ Response:
 [consumer-object]: /gateway/latest/admin-api/#consumer-object
 [acl-associating]: /plugins/acl/#associating-consumers
 
+
+### Behavior matrix
+
+The following table describes how {{site.base_gateway}} behaves in various scenarios:
+
+Request                        | Proxied to upstream service? | Response status code
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/key-auth/_index.md:400]: @jschmid1 For the entries in this column, does "has valid apikey" display exactly like that in the request? Or is this more of a description of what happens in the request? Ex. "The request has a valid API key".

File: app/_hub/kong-inc/key-auth/_index.md [hunk 3]
```diff
@@ -384,6 +392,19 @@ Response:
 [consumer-object]: /gateway/latest/admin-api/#consumer-object
 [acl-associating]: /plugins/acl/#associating-consumers
 
+
+### Behavior matrix
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/key-auth/_index.md:396]: @jschmid1 You mentioned wanting all plugins to have this. Did you mean just a behavior matrix in general or did you want us to pretty much copy/paste this exact table into each plugin doc?

Comment (jschmid1) [reply app/_hub/kong-inc/key-auth/_index.md:396]: just a behavior matrix in general. this needs to figured out for every plugin individually.

File: app/_hub/kong-inc/key-auth/_index.md (hunk 2 above)
Comment (jschmid1) [reply app/_hub/kong-inc/key-auth/_index.md:400]: This is more of a description. Feel free to adapt it

**Pull request merged**